docs/csharp/advanced-topics/interop/snippets/OfficeInterop/program.cs

```diff
@@ -48,39 +49,128 @@ static void Main(string[] args)
-            // Add has an optional parameter for specifying a particular template.
-            // Because no argument is sent in this example, Add creates a new workbook.
-            excelApp.Workbooks.Add();
+                // Create a new, empty workbook and add it to the collection returned
+                // by property Workbooks. The new workbook becomes the active workbook.
+                // Add has an optional parameter for specifying a particular template.
+                // Because no argument is sent in this example, Add creates a new workbook.
+                workbook = excelApp.Workbooks.Add();
 
-            // This example uses a single workSheet. The explicit type casting is
-            // removed in a later procedure.
-            Excel._Worksheet workSheet = (Excel.Worksheet)excelApp.ActiveSheet;
+                // This example uses a single workSheet. The explicit type casting is
+                // removed in a later procedure.
+                workSheet = (Excel.Worksheet)excelApp.ActiveSheet;
+                
+                // Establish column headings in cells A1 and B1.
+                workSheet.Cells[1, "A"] = "ID Number";
+                workSheet.Cells[1, "B"] = "Current Balance";
+
+                var row = 1;
+                foreach (var acct in accounts)
+                {
+                    row++;
+                    workSheet.Cells[row, "A"] = acct.ID;
+                    workSheet.Cells[row, "B"] = acct.Balance;
+                }
+
+                workSheet.Columns[1].AutoFit();
+                workSheet.Columns[2].AutoFit();
+
+                // Put the spreadsheet contents on the clipboard.
+                workSheet.Range["A1:B3"].Copy();
+                
+                // Save the workbook before closing
+                string fileName = System.IO.Path.Combine(
+                    Environment.GetFolderPath(Environment.SpecialFolder.Desktop), 
+                    "BankAccounts.xlsx");
+                workbook.SaveAs(fileName);
+            }
+            finally
+            {
+                // Clean up COM objects in reverse order of creation
+                if (workSheet != null)
+                {
+                    System.Runtime.InteropServices.Marshal.FinalReleaseComObject(workSheet);
+                    workSheet = null;
+                }
+                if (workbook != null)
+                {
+                    workbook.Close(true); // Save changes
+                    System.Runtime.InteropServices.Marshal.FinalReleaseComObject(workbook);
+                    workbook = null;
+                }
+                if (excelApp != null)
+                {
+                    excelApp.DisplayAlerts = true;
+                    excelApp.Quit();
+                    System.Runtime.InteropServices.Marshal.FinalReleaseComObject(excelApp);
+                    excelApp = null;
+                }
+                
+                // Force garbage collection
```

Comment:
.NET runtime does not guarantee that objects above gets collected. The JIT is free to extend lifetimes till end of the method. Assigning null to local variables like `excelApp = null;` may help, but it is not 100% reliable.

The only reliable way to get objects collected is by factoring out the code where it is not desirable to extend lifetimes into its own non-inlineable method.